portal-api/src/v1/validation/fields/currency-not-the-same-as-supply-contract-rules/currency.test.js

```diff
@@ -0,0 +1,66 @@
+const currencyRule = require('./currency');
+const { orderNumber } = require('../../../../utils/error-list-order-number');
+const { currencyIsDisabled } = require('../currency');
+
+jest.mock('../currency');
+
+describe('validation - currency', () => {
+  const errorList = {};
+
+  describe('when the facility has no currency object', () => {
+    it('should return validation error', () => {
+      const result = currencyRule({});
+
+      const expected = {
+        currency: {
+          order: orderNumber(errorList),
+          text: 'Enter the Currency',
+        },
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when the facility has no currency.id property', () => {
+    it('should return validation error', () => {
+      const mockFacility = {
+        currency: {},
+      };
+
+      const result = currencyRule(mockFacility);
+
+      const expected = {
+        currency: {
+          order: orderNumber(errorList),
+          text: 'Enter the Currency',
+        },
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
```

Comment:
I'd consider addressing this here -- We're setting ourselves up for some future tech debt on this test file as having the mocked function in this test now makes it so this suite cannot be reordered easily. If a dev adds a new test above this test, it'll behave different to below this test.

We should therefore either:
- Mock currencyIsDisabled everywhere, have a before each to clear the mock to clean up this config
- Spy on currencyIsDisabled just for this test, and have a cleanup for it after the assertion (mockRestore)
- Spy on it in the before each of `describe('when the facility has a currency that is disabled'` -- I prefer this as its related to the describe. We then clean up in the after each using a mock restore.


Also -- it's not async, so it should be `mockReturnValueOnce`

Reply:
`currencyIsDisabled` related tests (and related logic), is lacking.

This seems to be the only place where such tests exist because I added in this PR.

Reply:
Just fyi -- this is still susceptible to the above as it's only resetting the mockfn to be an empty mockfn -- we'd want `currencyIsDisabled.mockRestore();` in the after each.

I know this is very unlikely to be extended, but I tend to try and avoid leaving tests that _can_ leak in the future, as when you find them they can be a huge time sink to realise

Reply:
Updated to use `mockRestore`